### ISSUE TITLE
Ultra HDR support

### ISSUE THREAD

**Issue by gsdgethjfjj** [FossifyOrg/Gallery#166]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Gallery/issues) or [closed](https://github.com/FossifyOrg/Gallery/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Gallery/discussions) or [closed](https://github.com/FossifyOrg/Gallery/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Gallery/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Please add Ultra HDR photos support

### Why do you want this feature?

My Pixel phone can take Ultra HDR photos which looks great. Unfortunately Fossify gallery shows them as a regular SDR photos. 

### Additional information

These links might be helpful

https://developer.android.com/media/platform/hdr-image-format

https://developer.android.com/media/grow/ultra-hdr/display

**Comment by Aga-C** [comment]
Could you share a sample Ultra HDR photo for reference? Not every phone supports it, so it would be easier for developers.

**Comment by gsdgethjfjj** [comment]
![PXL_20240214_002410318~2](https://github.com/FossifyOrg/Gallery/assets/158708599/6f522f02-b58b-4b91-8eb5-e29992faae97)

**Comment by terminal-cs** [comment]
Any progress on this?

**Comment by naveensingh** [comment]
None yet.

**Comment by terminal-cs** [comment]
> None yet.

Unfortunate. This is the one thing stopping me from switching to this app from Google photos. I'd say I would help but really, I've never done any proper android app programming. No idea what it takes to get this working.

**Comment by BluePixel4k** [comment]
@naveensingh are you already working an this feature? :)

I think it would be good if the HDR support could be disabled/enabled via a setting.

**Comment by naveensingh** [comment]
> @naveensingh are you already working on this feature? :)

Among other things, yeah.

> I think it would be good if the HDR support could be disabled/enabled via a setting.

Why would you want that? It'll only be enabled for HDR images, so there will be no side effects when viewing regular files.

**Comment by BluePixel4k** [comment]
> Why would you want that? It'll only be enabled for HDR images, so there will be no side effects when viewing regular files.

Yes, but if a user has not only HDR images, the brightness would change constantly. This could be distracting for some users, maybe.

**Comment by Zarkorix** [comment]
I second the need to add a toggle to switch this off. I have a mixture of SDR and ultra HDR photos and it's now fairly jarring to swipe between them due to the brightness changes.

**Comment by Zarkorix** [comment]
I would say the other reason to have a toggle is to allow people to see what the photo will look like when shared/uploaded (since most platforms and devices still do not support ultra-HDR).

**Comment by Zarkorix** [comment]
@naveensingh I would say the other reason to have a toggle is to allow people to see what the photo will look like when shared/uploaded (since most platforms and devices still do not support ultra-HDR).

**Comment by naveensingh** [comment]
> I second the need to add a toggle to switch this off. I have a mixture of SDR and ultra HDR photos and it's now fairly jarring to swipe between them due to the brightness changes.

I do not see this effect on devices with HDR support. Could you guys share some sample images?

**Comment by Zarkorix** [comment]
@naveensingh Sorry - I'll clarify - there is no an issue with the phone's screen brightness increasing when rendering ultra-HDR images. Ultra-HDR images are, inherently, brighter and overwhelming in saturation. This means when flicking through a mixture of ultra-HDR and SDR images of the same scene (mixed together in an album), it can be jarring or distracting. 

In general though, I think the main argument for introducing a toggle to turn HDR rendering on/off is to allow users to see what an image will look like when shared (and thereby choose which images to share/upload to their social media platforms).

**Comment by naveensingh** [comment]
Got it. Please file a separate feature request for it. I'd like to see if this affects more people.

**Comment by Zarkorix** [comment]
Done - #564
